### ISSUE TITLE
Importing a PSModule dependency always imports the latest version

### ISSUE THREAD

**Issue by brantb** [RamblingCookieMonster/PSDepend#44]
I have a project which requires Pester 3.4.0 but my system has 4.x installed. Using `-Import` always imports the latest version, rather than the version specified in my dependencies.

## Steps to reproduce
```powershell
Get-Module -ListAvailable Pester
<# Output:
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     4.0.8      Pester                              {Describe, Context, It, Should...}
Script     3.4.0      Pester                              {Describe, Context, It, Should...}
#>
Remove-Module Pester -ea 0
Invoke-PSDepend -InputObject @{ Pester = '3.4.0' } -Import -Force
Get-Module Pester | Select Version
```

## Expected result
Pester 3.4.0 is imported.

## Actual result
Pester 4.0.8 is imported.

**Comment by RamblingCookieMonster** [comment]
Good catch! Will need to add some parameters to differentiate things [here](https://github.com/RamblingCookieMonster/PSDepend/blob/master/PSDepend/Private/Import-PSDependModule.ps1) and adjust various bad logic in the dependency script

Thanks!

**Comment by HeyItsGilbert** [comment]
Fixed in the current codebase. When a specific version is requested and installed, PSDependScripts/PSGalleryModule.ps1:232-238 matches it via Test-VersionEquality and imports that exact version; Private/Import-PSDependModule.ps1:16-21 passes it as RequiredVersion to Import-Module, so a newer side-by-side install is no longer picked up. Regression test: Tests/PSGalleryModule.Type.Tests.ps1, context "Multiple installed versions" (older version requested while newer installed → older imported), plus an equivalent assertion for PSResourceGet. Closing as completed.
